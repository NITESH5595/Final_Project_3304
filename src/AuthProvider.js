import React, { useEffect, useState, createContext } from "react";
import { onAuthStateChanged } from "firebase/auth";
import { auth } from "./base";

const AuthContext = createContext({});
const AuthProvider = ({ children }) => {
  const [currentUser, setCurrentUser] = useState(null);

  useEffect(() => {
    onAuthStateChanged(auth, (user) => {
      if (user) {
        setCurrentUser(user);
      } else {
        setCurrentUser(null);
      }
    });
    // console.log(currentUser);
  }, [currentUser]);

  return (
    <AuthContext.Provider value={{ currentUser }}>
      {children}
    </AuthContext.Provider>
  );
};
export { AuthContext, AuthProvider };
