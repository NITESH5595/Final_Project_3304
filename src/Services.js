import React, { useState } from "react";
import { signInWithEmailAndPassword } from "firebase/auth";
import "./index.css";
import { useNavigate } from "react-router-dom";
import { auth } from "./base";

function Services() {
  // const navigate = useNavigate();
  // const goBack = () => {
  //   navigate("/");
  // };
  return (
    <div className="mainContainer">
      <div className="loginForm">
        <p className="paragraph">This is our Services Page </p>
        <br />
        {/* <button className="btn" onClick={() => goBack()}>
          Go Back
        </button> */}
      </div>
    </div>
  );
}

export default Services;
