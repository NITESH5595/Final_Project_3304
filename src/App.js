import React from "react";
import { BrowserRouter, Routes, Route, Router } from "react-router-dom";
import Home from "./Home";
import Login from "./Login";
import SignUp from "./Signup";
import { AuthProvider } from "./AuthProvider";
import About from "./AboutUs";
import Services from "./Services";
import Customers from "./Customers";
import Weather from "./Weather";

function App() {
  return (
    <AuthProvider>
      <div className="app">
        <BrowserRouter>
          <Routes>
            <Route exact path="/" element={<Home />} />
            <Route exact path="/about" element={<About />} />
            <Route exact path="/services" element={<Services />} />
            <Route exact path="/customers" element={<Customers />} />
            <Route exact path="/weather" element={<Weather />} />
            <Route exact path="/login" element={<Login />} />
            <Route exact path="/signup" element={<SignUp />} />
          </Routes>
        </BrowserRouter>
      </div>
    </AuthProvider>
  );
}

export default App;
// npm run test -- --coverage --watchAll=false; for code coverage
