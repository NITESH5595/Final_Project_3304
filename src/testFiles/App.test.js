import React from "react";
import App from "../App";
import renderer from "react-test-renderer";
import { AuthProvider } from "../AuthProvider";
import { Services } from "../Services";

describe("App", () => {
  it("renders without crashing given the required props", () => {
    const wrapper1 = renderer.create(<App />).toJSON();
    expect(wrapper1).toMatchSnapshot();
    const wrapper = renderer.create(<AuthProvider />).toJSON();
    expect(wrapper).toMatchSnapshot();

    // const wrapper1 = shallow(<App />);

    // const wrapper2 = shallow(<Services />);

    // const wrapper3 = shallow(<Customers />);

    // expect(toJson(wrapper1)).toMatchSnapshot();

    // expect(toJson(wrapper2)).toMatchSnapshot();

    // expect(toJson(wrapper3)).toMatchSnapshot();
  });
});
