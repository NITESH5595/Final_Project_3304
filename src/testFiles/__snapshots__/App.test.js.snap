// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing given the required props 1`] = `
<div
  className="app"
>
  <div
    className="mainContainer"
  >
    <div
      className="buttons"
    >
      <button
        className="btn"
        onClick={[Function]}
      >
        Login
      </button>
      <button
        className="btn"
        onClick={[Function]}
      >
        Sign Up
      </button>
    </div>
  </div>
</div>
`;

exports[`App renders without crashing given the required props 2`] = `null`;
